be/src/olap/rowset/beta_rowset_writer.h

```diff
@@ -149,9 +157,15 @@ class BetaRowsetWriter : public RowsetWriter {
     std::vector<KeyBoundsPB> _segments_encoded_key_bounds;
     // record rows number of every segment
     std::vector<uint32_t> _segment_num_rows;
+
+    // ensure only one inflight segcompaction task for each rowset
     std::atomic<bool> _is_doing_segcompaction;
+    // enforce compare-and-swap on _is_doing_segcompaction
+    std::mutex _is_doing_segcompaction_lock;
```

Comment:
`_segcompacting_cond_lock` is enought to help to protect `_is_doing_segcompaction`, no need to import another lock?
I don't think we need two locks here, you can keep 1 of them, and the code can works well and simple?